src/Controls/src/Core/Page/Page.cs

```diff
@@ -449,6 +449,26 @@ protected virtual void LayoutChildren(double x, double y, double width, double h
 		/// </summary>
 		protected virtual void OnAppearing()
 		{
+			UpdateMenuBarItemBindingContext();
+		}
+
+		/// <summary>
+		/// Resets the BindingContext for each MenuBarItem when it becomes null after page navigation.
+		/// </summary>
+		private void UpdateMenuBarItemBindingContext()
```

Comment:
```suggestion
	 void UpdateMenuBarItemBindingContext()
```

please don t use private